cli/tests/unit/fetch_test.ts

```diff
@@ -23,12 +23,32 @@ unitTest({ perms: { net: true } }, async function fetchProtocolError(): Promise<
 unitTest(
   { perms: { net: true } },
   async function fetchConnectionError(): Promise<void> {
-    await assertThrowsAsync(
-      async (): Promise<void> => {
-        await fetch("http://localhost:4000");
-      },
-      TypeError,
-      "error trying to connect",
+    const initialPort = 4000;
+    const maxPort = 9999;
+    let port = initialPort;
+    let portFound = false;
+
+    while (port < maxPort) {
+      try {
+        const listener = Deno.listen({ port });
+        await assertThrowsAsync(
+          async (): Promise<void> => {
+            listener.close();
+            await fetch(`http://localhost:${port}`);
+          },
+          TypeError,
+          "error trying to connect",
+        );
+        portFound = true;
+        break;
+      } catch (e) {
+        port++;
+      }
```

Comment:
IMO, the port iteration logic kind of obscures that the purpose of the test is to run that fetch() command. I'd break it out into a helper function.

Another thing: the try/catch section is too big, it should only be around the `Deno.listen()` call.

Reply:
Thanks for the feedback. I've factored the referenced code into a helper function, and reduced the amount of logic in the try/catch. I think that it is now overall much more readable.